project/app.rb

```diff
@@ -0,0 +1,99 @@
+require_relative 'student'
+require_relative 'teacher'
+require_relative 'book'
+require_relative 'rental'
+
+class App
+  def initialize
+    @books = []
+    @rentals = []
+    @people = []
+  end
+
+  def list_books
+    puts '---------------------------------'
+    @books.each_with_index do |book, index|
+      puts "#{index + 1}. #{book.title} by #{book.author}"
+    end
+    puts '---------------------------------'
+  end
+
+  def list_people
+    puts '---------------------------------'
+    @people.each_with_index do |person, index|
+      puts "#{index + 1}. #{person.name} - #{person.class}"
+    end
+    puts '---------------------------------'
+  end
+
+  def create_student
+    print 'Age: '
+    age = gets.chomp.to_i
+    print 'Name: '
+    name = gets.chomp
+    print 'Has parent permission? (Y/N): '
+    permission = gets.chomp.downcase == 'y'
+    @people << if permission
+                 Student.new(age, name)
+               else
+                 Student.new(age, name, parent_permission: false)
+               end
+  end
+
+  def create_teacher
+    print 'Age: '
+    age = gets.chomp.to_i
+    print 'Name: '
+    name = gets.chomp
+    print 'Specialization: '
+    specialization = gets.chomp
+    @people << Teacher.new(age, specialization, name: name)
+  end
+
+  def create_person
+    puts 'Do you want to create a student (1) or a teacher (2)?'
+    person_type = gets.chomp.to_i
+    case person_type
+    when 1
+      create_student
+    when 2
+      create_teacher
+    else
+      puts 'Person type not recognized'
+    end
+    puts 'Person created successfully'
+    puts '---------------------------------'
+  end
+
+  def create_book
+    print 'Title: '
+    title = gets.chomp
+    print 'Author: '
+    author = gets.chomp
+    @books << Book.new(title, author)
+    puts 'Book created successfully!'
+    puts '---------------------------------'
+  end
+
+  def create_rental
+    puts 'Select a book from the following list by number'
+    list_books
+    book_index = gets.chomp.to_i - 1
+    puts 'Select a student from the following list by number (not id)'
+    list_people
+    person_index = gets.chomp.to_i - 1
+    puts 'Date: '
+    date = gets.chomp
+    @rentals << Rental.new(date, @people[person_index], @books[book_index])
+    puts 'Rental created successfully!'
+    puts '---------------------------------'
+  end
+
+  def list_rentals
+    puts '---------------------------------'
+    @rentals.each_with_index do |rental, index|
+      puts "#{index + 1}. #{rental.person.name} - #{rental.book.title} - #{rental.date}"
+    end
+    puts '---------------------------------'
+  end
```

Comment:
- [ ] Greetings! 📜🚀 As outlined in the project [guidelines](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_ui.md), it's important to include a method that enables users to list all rentals associated with a specific person ID. This enhancement will provide a valuable feature, allowing users to input their desired ID and retrieve a comprehensive list of their rentals. Please note that this functionality can only work effectively if you ensure that you keep track of each person's ID when creating them. To ensure your project aligns seamlessly with these project guidelines, let's go ahead and implement this functionality

```suggestion
  def list_rentals_for_person_id
    print "ID of person: "
    # *If needed, ask for parameters for the option.
    id = gets.chomp.to_i

    rentals = @rentals.filter {|rental| rental.person.id == id}
    puts "Rentals:"
    rentals.each do |rental|
      puts "Date: #{rental.date}, Book \"#{rental.book.title}\" by #{rental.book.author}"
    end
  end
```

Reply:
Thank you for your review! The files have all been moved into a single folder to avoid conflict and the rental method now works with the person ID.